main.go

```diff
@@ -190,11 +190,15 @@ func executeHTTPRequest(urlStr string, methodType string, uuid string, authKey s
 	resp, err := client.Do(req)
 
 	if err != nil {
-		return nil, resp.StatusCode, transactionID, fmt.Errorf("Error executing requests for url=%s, error=%v", urlStr, err)
+		return nil, http.StatusInternalServerError, transactionID, fmt.Errorf("Error executing requests for url=%s, error=%v", urlStr, err)
 	}
 
 	defer cleanUp(resp)
 
+	if resp == nil {
```

Comment:
Hm, this shouldn't happend according to the doc: https://golang.org/pkg/net/http/#Client.Do:
> If the returned error is nil, the Response will contain a non-nil Body...

Reply:
That's weird, 'cause I actually did this to test where the `panic` came from:

```

	resp, err := client.Do(req)

	if err != nil {
		appLog.Info(resp)
		return nil, resp.StatusCode, transactionID, fmt.Errorf("Error executing requests for url=%s, error=%v", urlStr, err)
	}
```

And it showed me a bunch of nils

Reply:
That's the case when err is not nil, so resp is expected to be nil